### ISSUE TITLE
Undefined method `io' for Minitest::Reporters::DelegateReporter

### ISSUE THREAD

**Issue by nickgnd** [seattlerb/minitest#594]
Hi, with both **ruby 2.2.3**(p173) and **ruby 2.2.4**(p178) I have this error if I try to benchmark an alghoritm. For example Fibonacci algorithm: https://gist.github.com/nickgnd/320498b5d4465e50222c
```bash
ruby -Ilib:test fibonacci_bench.rb              

Run options: --seed 1647

# Running:

/Users/Nico/.rvm/gems/ruby-2.2.3/gems/minitest-5.8.2/lib/minitest/benchmark.rb:22:in `run': undefined method `io' for #<Minitest::Reporters::DelegateReporter:0x007feb590e5780> (NoMethodError)
	from /Users/Nico/.rvm/gems/ruby-2.2.3/gems/minitest-5.8.2/lib/minitest.rb:155:in `block in __run'
	from /Users/Nico/.rvm/gems/ruby-2.2.3/gems/minitest-5.8.2/lib/minitest.rb:155:in `map'
	from /Users/Nico/.rvm/gems/ruby-2.2.3/gems/minitest-5.8.2/lib/minitest.rb:155:in `__run'
	from /Users/Nico/.rvm/gems/ruby-2.2.3/gems/minitest-5.8.2/lib/minitest.rb:129:in `run'
	from /Users/Nico/.rvm/gems/ruby-2.2.3/gems/minitest-5.8.2/lib/minitest.rb:56:in `block in autorun'
```

**The same benchmark with ruby 2.2.0 it works well.**
```bash
rvm use 2.2.0
Using /Users/Nico/.rvm/gems/ruby-2.2.0

ruby -Ilib:test fibonacci_bench.rb

Run options: --seed 64501

# Running:

bench_fib	 0.000599	 0.007859	 0.076222	 0.775600	 7.938904
.

Finished in 8.879690s, 0.1126 runs/s, 0.1126 assertions/s.

1 runs, 1 assertions, 0 failures, 0 errors, 0 skips
```

My RVM info:
```bash
ruby-2.2.3:

  system:
    uname:       "Darwin MacBook-Pro-di-Nicolo.local 14.3.0 Darwin Kernel Version 14.3.0: Mon Mar 23 11:59:05 PDT 2015; root:xnu-2782.20.48~5/RELEASE_X86_64 x86_64"
    system:      "osx/10.10/x86_64"
    bash:        "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)"
    zsh:         "/bin/zsh => zsh 5.0.5 (x86_64-apple-darwin14.0)"

  rvm:
    version:      "rvm 1.26.9 (latest) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]"
    updated:      "9 months 13 days 5 hours 8 minutes 56 seconds ago"
    path:         "/Users/Nico/.rvm"

  ruby:
    interpreter:  "ruby"
    version:      "2.2.3p173"
    date:         "2015-08-18"
    platform:     "x86_64-darwin14"
    patchlevel:   "2015-08-18 revision 51636"
    full_version: "ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin14]"

  homes:
    gem:          "/Users/Nico/.rvm/gems/ruby-2.2.3"
    ruby:         "/Users/Nico/.rvm/rubies/ruby-2.2.3"

  binaries:
    ruby:         "/Users/Nico/.rvm/rubies/ruby-2.2.3/bin/ruby"
    irb:          "/Users/Nico/.rvm/rubies/ruby-2.2.3/bin/irb"
    gem:          "/Users/Nico/.rvm/rubies/ruby-2.2.3/bin/gem"
    rake:         "/Users/Nico/.rvm/rubies/ruby-2.2.3/bin/rake"

  environment:
    PATH:         "/Users/Nico/.rvm/gems/ruby-2.2.3/bin:/Users/Nico/.rvm/gems/ruby-2.2.3@global/bin:/Users/Nico/.rvm/rubies/ruby-2.2.3/bin:/Users/Nico/.rvm/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/webdriver:/opt/X11/bin:/usr/local/mysql/bin:/Library/Frameworks/Python.framework/Versions/3.4/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/webdriver:/opt/X11/bin:/usr/local/mysql/bin"
    GEM_HOME:     "/Users/Nico/.rvm/gems/ruby-2.2.3"
    GEM_PATH:     "/Users/Nico/.rvm/gems/ruby-2.2.3:/Users/Nico/.rvm/gems/ruby-2.2.3@global"
    MY_RUBY_HOME: "/Users/Nico/.rvm/rubies/ruby-2.2.3"
    IRBRC:        "/Users/Nico/.rvm/rubies/ruby-2.2.3/.irbrc"
    RUBYOPT:      ""
    gemset:       ""
```

**Comment by zamith** [comment]
I tried it with 2.2.0, 2.2.2, 2.2.3 and 2.3.0-dev and they all work, both on version 5.8.2 and master of minitest. Not sure what it's happening on your machine.

However, if you have minitest installed as a gem, you don't need the `-I`, you should be able to run it with `ruby fibonacci_bench.rb`. If that doesn't work, please try getting minitest from github and give it a go.

**Comment by zenspider** [comment]
The bug is probably in `Minitest::Reporters::DelegateReporter`, which is not part of minitest.

**Comment by nickgnd** [comment]
Thanks for the replies. 
I think I've solved the problem. It was related at my machine. I uninstalled minitest gem ```gem uninstall minitest``` and then the benchmark works fine.

I also tried inside at a Rails project and as @zenspider said there is a compatibility problem with the gem ```minitest-reporters```. With this gem in my Gemfile the problem still remains (with ruby 2.2.3 and 2.2.4), independently if i require ```minitest-reporters``` in my ```test_helper``` file or not. Instead, when I remove it from my Gemfile and test_helper file, the benchmark works well.
